include/zephyr/drivers/fuel_gauge.h

```diff
@@ -173,8 +199,21 @@ struct fuel_gauge_set_property {
  * @return return=0 if successful, return < 0 if getting all properties failed, return > 0 if some
  * properties failed where return=number of failing properties.
  */
-typedef int (*fuel_gauge_get_property_t)(const struct device *dev,
-					 struct fuel_gauge_get_property *props, size_t props_len);
+__syscall int fuel_gauge_get_prop(const struct device *dev, struct fuel_gauge_get_property *props,
+				  size_t props_len);
+
+static inline int z_impl_fuel_gauge_get_prop(const struct device *dev,
+					     struct fuel_gauge_get_property *props,
+					     size_t props_len)
+{
+	const struct fuel_gauge_driver_api *api = (const struct fuel_gauge_driver_api *)dev->api;
+
+	if (api->get_property == NULL) {
+		return -ENOSYS;
```

Comment:
The errno errors are so odd, I usually use -ENOTSUP, but I'm happy with dealer's choice here.

Reply:
I was copying the pattern I saw from the `sensor` and `gpio` driver APIs.